### PR TITLE
API change the shape of the incidence curves

### PR DESCRIPTION
**Pull request by glemaitre**
Change some API such that the shape of returned arrays are:

- `predict_proba` is an array of shape (n_samples, n_events)
- `predict_cumulative_incidence` is an array of shape (n_samples, n_events, n_times)
- `predict_survival_function` is an array of shape (n_samples, n_times)

### CODE REVIEW

**Comment by ogrisel** [comment]
The goal is to be consistent with the conventions of scikit-learn where all `predict_` methods return datastructures with a first dimension that is aligned with the observations onto which the predictions are made.

File: hazardous/_survival_boost.py [hunk 1]
```diff
@@ -485,11 +483,23 @@ def predict_cumulative_incidence(self, X, times=None):
...
+            The time horizons at which to estimate the probabilities. If `None`, it uses
+            the grid generated during `fit` based on the parameter `n_time_grid_steps`.
+
         Returns
         -------
-        predicted_curves : array-like of shape (n_samples, n_events + 1)
-            The first column holds the survival probability to any event and others the
-            incicence probabilities for each event.
+        predicted_curves : ndarray of shape (n_samples, n_events + 1, n_times)
+            The estimated probabilities at different time horizons. The column
```
Comment (ogrisel) [inline hazardous/_survival_boost.py:498]: "column" is not meaningful any more for a 3D array.

The values for event with index 0 are the estimated probabilities of staying event-free at the requested time horizons for each observation described by the matching row of X. ...

File: hazardous/_survival_boost.py [hunk 2]
```diff
@@ -505,13 +515,29 @@ def predict_cumulative_incidence(self, X, times=None):
             predictions_at_t = self.estimator_.predict_proba(X_with_time)
             predictions_at_all_times.append(predictions_at_t)
 
-        predicted_curves = np.array(predictions_at_all_times).swapaxes(2, 0)
-
-        return predicted_curves
+        predicted_curves = np.array(predictions_at_all_times)
+        # roll axis to get a shape (n_samples, n_events + 1, n_times)
+        return np.rollaxis(predicted_curves, 1).swapaxes(1, 2)
```
Comment (ogrisel) [inline hazardous/_survival_boost.py:520]: Might be easier to follow with a single call to `np.transpose(predicted_curves, axes=(1, 2, 0))` instead.

**Review by ogrisel — APPROVED** [review]
Other than that LGTM.

File: hazardous/_survival_boost.py (hunk 2 above)
Comment (glemaitre) [inline hazardous/_survival_boost.py:520]:
```suggestion
        return np.transpose(predicted_curves, axes=(1, 2, 0))
```

File: hazardous/_survival_boost.py [hunk 3]
```diff
@@ -485,11 +483,23 @@ def predict_cumulative_incidence(self, X, times=None):
...
         -------
-        predicted_curves : array-like of shape (n_samples, n_events + 1)
-            The first column holds the survival probability to any event and others the
-            incicence probabilities for each event.
+        predicted_curves : ndarray of shape (n_samples, n_events + 1, n_times)
+            The estimated probabilities at different time horizons. The column
+            indexed 0 stores the estimated probabilities of staying event-free at
+            the requested time horizons for each observation described by the matching
+            row of X. The remaining columns store the estimated cumulated incidence
+            (or probability) for each event.
```
Comment (ogrisel) [inline hazardous/_survival_boost.py:502]:
```suggestion
            The estimated probabilities at different time horizons. The values at event
            index 0 are the estimated probabilities of staying event-free at
            the requested time horizons for each observation described by the matching
            row of X. The remaining event indices correspond to the estimated cumulated 
            incidence (or probability) for each event type.
```

**Review by ogrisel — APPROVED** [review]

**Pull request merged**